src/eventstore/EventStoreSubscriber.cs

```diff
@@ -185,27 +256,62 @@ public void Start()
+
+                        Task Appeared(
+                            StreamSubscription streamSubscription,
+                            ResolvedEvent e,
+                            CancellationToken cancellationToken) =>
+                            EventAppeared(e);
+
+                        switch (_liveOnly)
                         {
-                            var fromStream = _startingPosition.HasValue ?
-                                FromStream.After(new StreamPosition(_startingPosition.Value)) :
-                                FromStream.Start;
-
-                            _subscription = _eventStoreClient.SubscribeToStreamAsync(
-                                   _streamName,
-                                   fromStream,
-                                   (s, e, t) => EventAppeared(e),
-                                   resolveLinkTos: true,
-                                   subscriptionDropped: SubscriptionDropped).Result;
+                            case true when !_subscribeToAll:
+                                _subscription = _eventStoreClient.SubscribeToStreamAsync(
+                                    _streamName,
+                                    FromStream.End,
+                                    Appeared,
+                                    resolveLinkTos,
+                                    SubscriptionDropped).Result;
+                                break;
+                            case false when !_subscribeToAll:
+                            {
+                                var fromStream = _startingPosition.HasValue ?
+                                    FromStream.After(new StreamPosition(_startingPosition.Value)) :
+                                    FromStream.Start;
+
+                                _subscription = _eventStoreClient.SubscribeToStreamAsync(
+                                    _streamName,
+                                    fromStream,
+                                    Appeared,
+                                    resolveLinkTos,
+                                    SubscriptionDropped).Result;
+                                break;
+                            }
+
+                            case true when _subscribeToAll:
+                                _subscription = _eventStoreClient.SubscribeToAllAsync(
+                                        FromAll.End,
+                                        Appeared,
+                                        resolveLinkTos,
+                                        SubscriptionDropped,
+                                        filterOptions)
+                                    .Result;
+                                break;
+                            case false when _subscribeToAll:
+                                var fromAll = _startingPosition.HasValue ?
+                                    FromAll.After(new Position(_startingPosition.Value, _startingPosition.Value)) :
```

Comment:
Have we verified that prepare and commit positions are always equal? If not we will need to store both in our checkpoint.

Reply:
Discussed in Teams (https://teams.microsoft.com/l/message/19:2c1ca90124fc455ea75b289c3ad3c3ec@thread.skype/1670234344140?tenantId=b0c1be47-abd8-448c-a6e0-b1ef65a16eae&groupId=ee74f8db-6a44-4230-9890-0e2277229c1f&parentMessageId=1669294562648&teamName=Corsham%20Science%20-%20Software%20Development%20Team&channelName=General&createdTime=1670234344140&allowXTenantAccess=false). Added lots of events and ran a catch up that would throw if these positions weren't the same.